### PR TITLE
Review Interaction - Likes and Dislikes Feature

### PR DESCRIPTION
**Pull request by Armetiis**
### Feature Overview
This PR introduces a **user interaction system** for reviews, allowing users to express their opinions through likes and dislikes. The new feature provides more dynamic and engaging review experiences.

image
image

### Key Changes
- **Interactive Review Feedback**: Users can now like or dislike reviews through Thumbs Up, respectively Thumbs down icon buttons
- **Smart Interaction Handling**: Prevents simultaneous likes and dislikes
- **Counters with Immediate Visual Feedback:** Users can see a counter for likes and dislikes that is instantly updated upon their interaction
- **New Sorting Options**: New review sorting methods based on interactions

### ViewModel Implementation Details
- Created private interaction methods:
  - `addLikeToReview` (which also removes a dislike if there is one from that user)
  - `removeLikeFromReview`
  - `addDislikeToReview` (which also removes a like if there is one from that user)
  - `removeDislikeFromReview`
- Developed `handleInteraction` that given a Review, a userId and a Boolean (like/dislike), can assess which of the above to call, making therefore the UI very simple and allowing
- Ensured the `updateReview` method also updates the State holding reviews for a parking, allowing the update to be reflected in real time.

### Coverage
image

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/cyrcle/ui/review/AllReviewsScreenTest.kt [hunk 1]
```diff
@@ -182,6 +186,90 @@ class AllReviewsScreenTest {
...
+
+    // We check that the first review is still first
+    composeTestRule.onNodeWithTag("LikeCount0", useUnmergedTree = true).assertTextEquals("1")
+    composeTestRule.onNodeWithTag("DislikeCount0", useUnmergedTree = true).assertTextEquals("0")
+    composeTestRule.onNodeWithTag("LikeCount1", useUnmergedTree = true).assertTextEquals("0")
+    composeTestRule.onNodeWithTag("DislikeCount1", useUnmergedTree = true).assertTextEquals("0")
+  }
+
+  @Test
+  fun sortReviewsByInteractionTest() {
```
Comment (matttsss) [inline app/src/androidTest/java/com/github/se/cyrcle/ui/review/AllReviewsScreenTest.kt:267]: Is it normal to have an empty test here?

**Review by matttsss — COMMENTED** [review]
Honestly a functional and clean code, good job!

**Review by sean-prz — APPROVED** [review]
I have nothing the add, the code is really clean making it easily understandable.
And the feature works as expected.

Really, Great work !

**Review by matttsss — APPROVED** [review]
Thanks for the change!

**Pull request merged**